### ISSUE TITLE
Environment variables not reset between classes

### ISSUE THREAD

**Issue by MarkRx** [stefanbirkner/system-rules#61]
Environment variables should be reset after each class. Otherwise setup from one class leaks into another class which causes inconsistent behavior.

Example:

- Run A.java alone. It passes
- Run B.java alone. It passes
- Run A.java then B.java in a single execution. It fails. The environment variable "A" set in A.java leaks into the B.java test class.

A.java:
```
public class A {
	  @Rule
	  public EnvironmentVariables environmentVariables = new EnvironmentVariables();
	  {
		  environmentVariables.set("A", "A");
	  };
	  
	  @Test
	  public void test1() {
		  assertEquals("A", System.getenv("A"));
		  assertEquals(null, System.getenv("B"));
	  }
}
```

B.java:
```
public class B {
	  @Rule
	  public EnvironmentVariables environmentVariables = new EnvironmentVariables();
	  {
		  environmentVariables.set("B", "B");
	  };
	  
	  @Test
	  public void test1() {
		  assertEquals(null, System.getenv("A"));
		  assertEquals("B", System.getenv("B"));
	  }
}
```

**Comment by stefanbirkner** [comment]
Thanks for reporting the bug. I can reproduce it and work on a fix.

**Comment by stefanbirkner** [comment]
Currently the methods of `EnvironmentVariables` are not designed to be used outside of an `@Before` or the test itself. If you put the `environmentVariables.set("B", "B")` into an  `@Before` method then everything works fine.

Nevertheless I consider the current behaviour as a bug and will fix it.

**Comment by stefanbirkner** [comment]
This is fixed with System Rules 1.18.0: 063b1c82864c11b9dd3e14846665f796f340b7d5
